### PR TITLE
Add not authorized condition for stream error

### PR DESCRIPTION
**Pull request by henry61024**
Support `not-authorized` condition for `stream:error`.

This allow us to distinguish `stream:error` conditions more specifically.

Please refer to https://xmpp.org/rfcs/rfc6120.html#streams-error-conditions-not-authorized.

### CODE REVIEW

**Comment by jlaine** [comment]
Could you add some context about how this condition is encountered? Usually the "not-authorized" comes as part of the SASL exchange.

Also: do we really need the "bad-auth" case you added? It was added in the SASL handling as a workaround for faulty servers, I'd like to know for sure we **need** it at the stream level before adding it too.

**Comment by henry61024** [comment]
Because we want to mange our user accounts with features such as remove/lockout users.

When a user has been remove/lockout, xmpp server([Openfire](https://www.igniterealtime.org/projects/openfire/)) will send a `not-authorized` package in order to kickoff the unauthorized user. So our clients also need to support the `not-authorized` package.

About the `bad-auth`: I'm just referring to the error condition in

 https://github.com/qxmpp-project/qxmpp/blob/ba93161573503c666df1e6ba141e157ecdc6fd27/src/client/QXmppOutgoingClient.cpp#L614

**Comment by lnjX** [comment]
Don't we need to add all the other error cases, too?

File: src/client/QXmppOutgoingClient.cpp [hunk 1]
```diff
@@ -563,6 +563,8 @@ void QXmppOutgoingClient::handleStanza(const QDomElement &nodeRecv)
 
         if (!nodeRecv.firstChildElement("conflict").isNull())
             d->xmppStreamError = QXmppStanza::Error::Conflict;
+        else if (!nodeRecv.firstChildElement("not-authorized").isNull() || !nodeRecv.firstChildElement("bad-auth").isNull())
```
Comment (lnjX) [inline src/client/QXmppOutgoingClient.cpp]: The 'bad-auth' thing was a workaround for (old) prosody servers. Could you remove that. I think we can merge it then.

Comment (henry61024) [reply src/client/QXmppOutgoingClient.cpp]: Done. Fixed in b2bcfd9

**Pull request merged**